src/NServiceBus.FileBasedRouting/XmlRoutingFileAccess.cs

```diff
@@ -1,27 +1,23 @@
 ﻿namespace NServiceBus.FileBasedRouting
 {
     using System;
-    using System.IO;
     using System.Xml;
     using System.Xml.Linq;
 
     class XmlRoutingFileAccess
     {
-        public string FilePath { get; }
+        public Uri FileUri { get; }
 
-        public XmlRoutingFileAccess(string filePath)
+        public XmlRoutingFileAccess(Uri fileUri)
         {
-            FilePath = filePath;
+            FileUri = fileUri;
         }
 
         public XDocument Read()
         {
             try
             {
-                using (var fileStream = File.OpenRead(FilePath))
-                {
-                    return XDocument.Load(fileStream);
-                }
+                return XDocument.Load(FileUri.ToString());
```

Comment:
Another thing: can this become a security problem (loading XML from a URL)? I guess if somebody has rights to change the config of an endpoint to provide his URL, he can do whatever he wants. A more likely situation would a some kind of men-in-the-middle. Should we enforce HTTPS?